### PR TITLE
Validate auth_backend path, disallowing leading/trailing '/'s

### LINKED ISSUE #865: vault_auth_backend provider produces inconsistent result
When using the `vault_auth_backend` provider, if you give it a `path` that begins with a `/`, it fails during apply with an inconsistent result. However, it does not fail during plan.

### Terraform Version

0.13.3

### Affected Resource(s)
Please list the resources as a list, for example:

- vault_auth_backend

### Terraform Configuration Files
```hcl
resource "vault_auth_backend" "kubernetes" {
  type        = "kubernetes"
  path        = "/path/with/slash/prefix"
}
```

### Debug Output

https://gist.github.com/dhdersch/1b7a93af1b3db314a1cf491d70c6155d

### Panic Output
No panic produced

### Expected Behavior
What should have happened?

I think that Terraform should have informed me that beginning with a / is invalid.
### Actual Behavior
What actually happened?

The plan worked, but the apply failed with the following error:

https://gist.github.com/dhdersch/1b7a93af1b3db314a1cf491d70c6155d

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan` - this will work
1. `terraform apply` - this will fail

### Important Factoids

None

### References

None

### PR DESCRIPTION
**Pull request by benashz**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #865 

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note

```

Output from acceptance testing:

```
$ time make testacc TESTARGS='-v -test.count 1 -test.run TestResourceAuth*' 

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test -v -v -test.count 1 -test.run TestResourceAuth* -timeout 30m ./...
PASS
ok      github.com/hashicorp/terraform-provider-vault/util      0.265s [no tests to run]
=== RUN   TestResourceAuth
--- PASS: TestResourceAuth (3.30s)
=== RUN   TestResourceAuthTune
--- PASS: TestResourceAuthTune (2.66s)
PASS
ok      github.com/hashicorp/terraform-provider-vault/vault     7.822s
make testacc TESTARGS='-v -test.count 1 -test.run TestResourceAuth*'  23.85s user 11.51s system 306% cpu 11.540 total
...
```

### CODE REVIEW

File: website/docs/r/auth_backend.html.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ The following arguments are supported:
 
 * `type` - (Required) The name of the auth method type
 
-* `path` - (Optional) The path to mount the auth method — this defaults to the name of the type
+* `path` - (Optional) The path to mount the auth method — this defaults to the name of the type.
```
Comment (vinay-gopalan) [inline website/docs/r/auth_backend.html.md:31]: nit: while we're here could we also update the docs for `type` and `description` to also have periods at the end for consistency? No big deal though

**Review by vinay-gopalan — APPROVED** [review]
LGTM!

File: website/docs/r/auth_backend.html.md (hunk 1 above)
Comment (benashz) [reply website/docs/r/auth_backend.html.md:31]: Sounds good.

Comment (benashz) [reply website/docs/r/auth_backend.html.md:31]: Fixed with https://github.com/hashicorp/terraform-provider-vault/pull/1471/commits/c31cb428d18b64ac5519563bb1ec37f9aee6e2fd

**Pull request merged**